src/main/java/be/seeseemelk/mockbukkit/scoreboard/TeamMock.java

```diff
@@ -52,107 +56,109 @@ public TeamMock(String name, ScoreboardMock board)
 	public void displayName(@Nullable Component displayName) throws IllegalStateException, IllegalArgumentException
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		checkRegistered();
+		this.displayName = displayName == null ? Component.empty() : displayName;
 	}
 
 	@Override
 	public @NotNull Component prefix() throws IllegalStateException
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		checkRegistered();
+		return this.prefix;
 	}
 
 	@Override
 	public void prefix(@Nullable Component prefix) throws IllegalStateException, IllegalArgumentException
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		checkRegistered();
+		this.prefix = prefix == null ? Component.empty() : prefix;
 	}
 
 	@Override
 	public @NotNull Component suffix() throws IllegalStateException
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		checkRegistered();
+		return this.suffix;
 	}
 
 	@Override
 	public void suffix(@Nullable Component suffix) throws IllegalStateException, IllegalArgumentException
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		checkRegistered();
+		this.prefix = suffix == null ? Component.empty() : suffix;
 	}
 
 	@Override
 	public boolean hasColor()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		checkRegistered();
+		return this.color.isColor();
 	}
 
 	@Override
 	public @NotNull TextColor color() throws IllegalStateException
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		Preconditions.checkState(hasColor(), "Team colors must have hex values");
+		return TextColor.color(this.color.asBungee().getColor().getRGB());
 	}
 
 	@Override
 	public void color(@Nullable NamedTextColor color)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		checkRegistered();
+		this.color = color == null ? ChatColor.RESET : ChatColor.valueOf(color.toString().toUpperCase(Locale.ROOT));
 	}
 
 	@Override
 	public @NotNull String getDisplayName() throws IllegalStateException
 	{
 		checkRegistered();
-		return this.displayName;
+		return LegacyComponentSerializer.legacySection().serialize(this.displayName);
 	}
 
 	@Override
-	public void setDisplayName(String s)
+	public void setDisplayName(@NotNull String displayName)
 	{
+		Preconditions.checkNotNull(displayName, "Display name cannot be null");
 		checkRegistered();
-		this.displayName = s;
+		this.displayName = LegacyComponentSerializer.legacySection().deserialize(displayName);
 	}
 
 	@Override
 	public @NotNull String getPrefix() throws IllegalStateException
 	{
 		checkRegistered();
-		return this.prefix;
+		return LegacyComponentSerializer.legacySection().serialize(this.prefix);
 	}
 
 	@Override
-	public void setPrefix(String s)
+	public void setPrefix(@NotNull String prefix)
```

Comment:
String display names cannot be longer than 128 characters, excluding colors.

Reply:
The limit is actually 128/64/64 (display name/prefix/suffix) and is not really sync with vanilla, but it makes sense to match the checks made by Bukkit.